### PR TITLE
Added server connector builder

### PR DESCRIPTION
**Pull request by ChrisRo89**
Created builder out of the duplicated code and refactored Http/HttpsFactory

### CODE REVIEW

**Comment by ChrisRo89** [comment]
@jglick Are you now happy? :-D

File: src/main/java/winstone/ConnectorFactory.java [hunk 1]
```diff
@@ -24,4 +24,5 @@
      * @param server
      */
     boolean start(Map args, Server server) throws IOException;
-}
+
+}
```
Comment (jglick) [inline src/main/java/winstone/ConnectorFactory.java]: Revert changes to this file please.

File: src/main/java/winstone/ServerConnectorBuilder.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
+package winstone;
+
+import org.eclipse.jetty.server.ForwardedRequestCustomizer;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.HttpConnectionFactory;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+
+public class ServerConnectorBuilder {
```
Comment (jglick) [inline src/main/java/winstone/ServerConnectorBuilder.java]:
```suggestion
class ServerConnectorBuilder {
```

File: src/main/java/winstone/ServerConnectorBuilder.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    public ServerConnectorBuilder withRequestHeaderSize(int requestHeaderSize) {
+        this.requestHeaderSize = requestHeaderSize;
+        return this;
+    }
+
+    public ServerConnector build() {
+
+        ServerConnector sc;
+
+        if(sslContextFactory != null) {
```
Comment (jglick) [inline src/main/java/winstone/ServerConnectorBuilder.java]:
```suggestion
        if (sslContextFactory != null) {
```

File: src/main/java/winstone/ServerConnectorBuilder.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            sc = new ServerConnector(server, acceptors, selectors, sslContextFactory);
+        }
+        else {
```
Comment (jglick) [inline src/main/java/winstone/ServerConnectorBuilder.java]:
```java
} else {
```

File: src/main/java/winstone/ServerConnectorBuilder.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    private String listenerAddress;
+    private Server server;
+    private SslContextFactory sslContextFactory;
+
+    public ServerConnectorBuilder withListenerPort(int listenerPort) {
+        this.listenerPort = listenerPort;
+        return this;
+    }
+
+    public ServerConnectorBuilder withKeepAliveTimeout(int keepAliveTimeout) {
```
Comment (jglick) [inline src/main/java/winstone/ServerConnectorBuilder.java]: AFAICT this is not being called?

Comment (ChrisRo89) [reply src/main/java/winstone/ServerConnectorBuilder.java]: Thank you for the hint, I forgot to add the parameter there.

File: .idea/.gitignore [hunk 6]
```diff
@@ -0,0 +1,2 @@
+# Default ignored files
+/workspace.xml
```
Comment (res0nance) [inline .idea/.gitignore]: The `.gitignore` file should ignore `.idea` instead of including this file

**Comment by oleg-nenashev** [comment]
I am taking another delay because of https://issues.jenkins-ci.org/browse/JENKINS-60409 and (https://issues.jenkins-ci.org/browse/JENKINS-60199 which may need backporting. Thanks for your patience.

File: src/main/java/winstone/HttpConnectorFactory.java [hunk 7]
```diff
@@ -28,31 +24,22 @@
...
-    }
+        else {
+            ServerConnectorBuilder scb = new ServerConnectorBuilder()
+                .withServer(server)
+                .withAcceptors(Option.JETTY_ACCEPTORS.get(args))
+                .withSelectors(Option.JETTY_SELECTORS.get(args))
+                .withListenerPort(listenPort)
+                .withListenerAddress(Option.HTTP_LISTEN_ADDRESS.get(args))
+                .withRequestHeaderSize(Option.REQUEST_HEADER_SIZE.get(args))
+                .withKeepAliveTimeout(Option._KEEP_ALIVE_TIMEOUT.get(args));
```
Comment (jglick) [inline src/main/java/winstone/HttpConnectorFactory.java:39]: #100 FTR

Comment (olamy) [reply src/main/java/winstone/HttpConnectorFactory.java:39]: @jglick This is what I mean it has been introduced by mistake during this change

File: src/main/java/winstone/HttpConnectorFactory.java [hunk 8]
```diff
@@ -28,31 +24,22 @@
     public boolean start(Map args, Server server) throws IOException {
         // Load resources
         int listenPort = Option.HTTP_PORT.get(args);
-        String listenAddress = Option.HTTP_LISTEN_ADDRESS.get(args);
-        int keepAliveTimeout = Option.HTTP_KEEP_ALIVE_TIMEOUT.get(args);
```
Comment (olamy) [inline src/main/java/winstone/HttpConnectorFactory.java:32]: @jglick ^

File: src/main/java/winstone/HttpConnectorFactory.java (hunk 7 above)
Comment (jglick) [reply src/main/java/winstone/HttpConnectorFactory.java:39]: The _misuse_ of this option was introduced with this change. But the actual option has always (long?) been there. We need to remove the option IMO.

**Pull request merged**